src/arrayMethodJoin.js

```diff
@@ -6,7 +6,47 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    const delimiter = getDelimiter(separator);
+
+    let output = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const current = isValidItem(this[i])
+        ? this[i]
+        : '';
+      const part = !isLastIndex(this, i)
+        ? current + delimiter
+        : current;
+
+      output += part;
+    }
+
+    return output;
   };
 }
 
+function getDelimiter(separator) {
+  if (separator) {
+    return separator.toString();
+  }
+
+  if (separator === '') {
+    return separator;
+  }
+
+  if (separator === null) {
+    return 'null';
+  }
```

Comment:
No need for such a big check, because if you add something to string it become a string 
![image](https://github.com/mate-academy/js_array-method-join/assets/46566190/4f8970dc-6da3-4436-9c13-f0b8bf6dc35f)
![image](https://github.com/mate-academy/js_array-method-join/assets/46566190/a7ea2797-b908-4bd0-9ae4-cfde955d3004)